extensions/lifecycle/tasks/LifecycleDeleteObjectTask.js

```diff
@@ -117,13 +111,27 @@ class LifecycleDeleteObjectTask extends BackbeatTask {
         let reqMethod;
 
         const actionType = entry.getActionType();
+        let req = null;
         if (actionType === 'deleteObject') {
             reqMethod = 'deleteObject';
+            const backbeatClient = this.getBackbeatClient(accountId);
+            if (!backbeatClient) {
+                log.error('failed to get backbeat client', { accountId });
+                return done(errors.InternalError
+                    .customizeDescription('Unable to obtain client'));
+            }
+            req = backbeatClient.deleteObjectFromExpiration(reqParams);
         } else if (actionType === 'deleteMPU') {
             reqParams.UploadId = entry.getAttribute('details.UploadId');
             reqMethod = 'abortMultipartUpload';
+            const s3Client = this.getS3Client(accountId);
+            if (!s3Client) {
+                log.error('failed to get S3 client', { accountId });
+                return done(errors.InternalError
+                    .customizeDescription('Unable to obtain client'));
+            }
+            req = s3Client[reqMethod](reqParams);
```

Comment:
I believe that eventually the "abortMPU" triggered by Lifecycle would also need its own origin op? Maybe comment here if that's indeed the case.

Reply:
There is no lifecycle specific event for abortMPU according to AWS documentation https://docs.aws.amazon.com/AmazonS3/latest/userguide/notification-how-to-event-types-and-destinations.html

The s3 call here already sets the originOP accordingly so we don't need to do anything specific for this case

Reply:
There is no lifecycle specific event for abortMPU according to AWS documentation https://docs.aws.amazon.com/AmazonS3/latest/userguide/notification-how-to-event-types-and-destinations.html

The s3 call here already sets the originOP accordingly so we don't need to do anything specific for this case